pytorch_toolkit/object_detection/tests/common/test_case.py

```diff
@@ -260,3 +299,109 @@ def test_alt_ssd_export_on_cpu(self):
+            src_template_folder = os.path.join(cls.templates_folder, 'object_detection', problem_name, model_name)
+
+            cls.skip_non_instantiated_template_if_its_allowed(src_template_folder)
+            cls.src_template_folder = src_template_folder
+
+            cls.template_folder = cls.generate_template_folder_name(cls.src_template_folder,
+                                                                    cls.template_updates_description)
+            cls.copy_template_folder(cls.src_template_folder, cls.template_folder)
+
+            cls.template_file = os.path.join(cls.template_folder, 'template.yaml')
+            cls.apply_update_dict_params_to_template_file(cls.template_file, template_update_dict)
+
+            cls.ann_file = ann_file
+            cls.img_root = img_root
+            cls.dependencies = get_dependencies(cls.template_file)
+
+            download_snapshot_if_not_yet(cls.template_file, cls.template_folder)
+
+            run_through_shell(
+                f'cd {cls.template_folder};'
+                f'pip install -r requirements.txt;'
+            )
+            logging.info(f'End setting up class for {problem_name}/{model_name}, {cls.template_updates_description}')
+
+        @staticmethod
+        def generate_template_updates_description(template_update_dict):
+            keys = sorted(template_update_dict.keys())
+            template_updates_description = "_".join(k+"="+str(template_update_dict[k]) for k in keys)
+            return template_updates_description
+
+        @staticmethod
+        def generate_template_folder_name(src_template_folder, template_updates_description):
+            assert not src_template_folder.endswith('/')
+            template_folder = src_template_folder + '__' +  template_updates_description
+            return template_folder
+
+        @staticmethod
+        def copy_template_folder(src_template_folder, template_folder):
+            logging.info(f'Copying {src_template_folder} to {template_folder}')
+            run_through_shell(f'cp -a "{src_template_folder}" "{template_folder}"')
+            assert os.path.isdir(template_folder), f'Cannot create {template_folder}'
+
+        @staticmethod
+        def apply_update_dict_params_to_template_file(template_file, update_params):
+            template_data = mmcv.Config.fromfile(template_file)
+            template_data.dump(template_file + '.backup.yaml')
+
+            template_data.merge_from_dict(update_params)
+            template_data.dump(template_file)
+
+        def skip_if_cpu_is_not_supported(self):
+            with open(self.template_file) as read_file:
+                training_targets = [x.lower() for x in yaml.load(read_file, yaml.SafeLoader)['training_target']]
+            if 'cpu' not in training_targets:
+                self.skipTest('CPU is not supported.')
+
+        @staticmethod
+        def skip_non_instantiated_template_if_its_allowed(template_folder):
```

Comment:
please extract this method out of the class

Reply:
Ok

Reply:
Done